python/heterocl/tvm/schedule.py

```diff
@@ -334,36 +334,42 @@ def reuse_at(self, target, parent, axis, name):
     def partition(self, target, partition_type, dim, factor):
         return _api_internal._SchedulePartition(self, target, dim, factor, partition_type)
 
-    def to(self, tensor, dst, src, index=0,
-           types=_expr.StreamExpr.Channel, depth=1):
+    def join(self, target, src, dst=None, types=_expr.StreamExpr.FIFO, depth=1):
+        """ join multiple writes to target tensor """
+        return _api_internal._ScheduleJoin(self, target, src, dst, types, depth)
+
+    def to(self, tensor, dst, src, axis=0,
+           types=_expr.StreamExpr.FIFO, depth=1):
```

Comment:
Can we use `type` instead of `types`? Also can we have the same order for `src` and `dst`?

Reply:
Sure. But `join` is from sources to destination. Having dest before src looks weird to me.

Reply:
Then `to` should also be from source to destination?

Reply:
for data movement between host-device, it's from target to dest. for data movement between stages, yes, we should change to src to dest.